pkg/skaffold/survey/survey.go

```diff
@@ -103,7 +131,49 @@ func (s *Runner) OpenSurveyForm(_ context.Context, out io.Writer, id string) err
 		logrus.Debugf("could not open url %s", sc.URL)
 		return err
 	}
-	// Currently we will only update the global config 	survey taken
-	// When prompting for the survey, we need to use the same field.
-	return sConfig.UpdateHaTSSurveyTaken(s.configFile)
+
+	if id == HatsID {
+		// Currently we will only update the global config survey taken
+		// When prompting for the survey, we need to use the same field.
+		return sConfig.UpdateHaTSSurveyTaken(s.configFile)
+	}
+	return sConfig.UpdateUserSurveyTaken(s.configFile, id)
+}
+
+func surveysTaken(sc *sConfig.SurveyConfig) map[string]struct{} {
+	if sc == nil {
+		return map[string]struct{}{}
+	}
+	taken := map[string]struct{}{}
+	if timeutil.LessThan(sc.LastTaken, 90*24*time.Hour) {
+		taken[HatsID] = struct{}{}
+	}
+	for _, cs := range sc.UserSurveys {
+		if *cs.Taken {
+			taken[cs.ID] = struct{}{}
+		}
+	}
+	return taken
+}
+
+func (s *Runner) presentSurvey(gc *sConfig.SurveyConfig) string {
+	taken := surveysTaken(gc)
+	var candidates []config
+	for _, sc := range surveys {
+		if _, ok := taken[sc.id]; !ok && sc.isActive() {
```

Comment:
ok doesn't capture the meaning here.
```suggestion
		if _, found := taken[sc.id]; !found && sc.isActive() {
```

Even better might be to rename `taken` -> `takenSurveys` and then use `if _, taken := takenSurveys[sc.id]; !taken && sc.isActive() {`